tensorflow/core/util/gpu_kernel_helper.h

```diff
@@ -13,34 +13,77 @@ See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
 
-#ifndef TENSORFLOW_CORE_UTIL_CUDA_KERNEL_HELPER_H_
-#define TENSORFLOW_CORE_UTIL_CUDA_KERNEL_HELPER_H_
+#ifndef TENSORFLOW_CORE_UTIL_GPU_KERNEL_HELPER_H_
+#define TENSORFLOW_CORE_UTIL_GPU_KERNEL_HELPER_H_
 
-#if GOOGLE_CUDA
+#if GOOGLE_CUDA || TENSORFLOW_USE_ROCM
 
-#include "tensorflow/core/util/cuda_device_functions.h"
-#include "tensorflow/core/util/cuda_launch_config.h"
+#include "tensorflow/core/util/gpu_device_functions.h"
+#include "tensorflow/core/util/gpu_launch_config.h"
 
+#if GOOGLE_CUDA
 #include "cuda/include/cuda_fp16.h"
+#endif
+
+#if GOOGLE_CUDA
+#define TF_RED_WARPSIZE 32
+#elif TENSORFLOW_USE_ROCM
+#define TF_RED_WARPSIZE 64
+#endif
+
+#if GOOGLE_CUDA
+#define GPU_LAUNCH_KERNEL(kernel, block_count, threads_per_block, \
+                          shared_mem, stream, ...) \
+  TF_CHECK_OK(CudaLaunchKernel(kernel, block_count, threads_per_block, \
+                               shared_mem, stream, __VA_ARGS__));
+#elif TENSORFLOW_USE_ROCM
+#define GPU_LAUNCH_KERNEL(kernel, block_count, threads_per_block, \
+                          shared_mem, stream, ...) \
+  hipLaunchKernelGGL(kernel, \
+    block_count, threads_per_block, shared_mem, stream, \
+    __VA_ARGS__);
+#endif
+
+// Deprecated, use 'for(int i : GpuGridRangeX(n))' instead.
+#define GPU_1D_KERNEL_LOOP(i, n) \
+  for (int i : ::tensorflow::GpuGridRangeX<int>(n))
+// Deprecated, use 'for(int i : GpuGridRange?(n))' instead.
+#define GPU_AXIS_KERNEL_LOOP(i, n, axis) \
+  for (int i : ::tensorflow::GpuGridRange##axis<int>(n))
 
-// Deprecated, use 'for(int i : CudaGridRangeX(n))' instead.
-#define CUDA_1D_KERNEL_LOOP(i, n) \
-  for (int i : ::tensorflow::CudaGridRangeX<int>(n))
-// Deprecated, use 'for(int i : CudaGridRange?(n))' instead.
-#define CUDA_AXIS_KERNEL_LOOP(i, n, axis) \
-  for (int i : ::tensorflow::CudaGridRange##axis<int>(n))
+#if GOOGLE_CUDA
+#define gpuSuccess cudaSuccess
+#define GPUGETERRORSTRING(error) cudaGetErrorString(error)
+using gpuStream_t = cudaStream_t;
+#define GPUGETLASTERROR() cudaGetLastError()
+using gpuError_t = cudaError_t;
+#define GPUSUCCESSS cudaSuccess
+#elif TENSORFLOW_USE_ROCM
```

Comment:
Please add some whitespace to make this easy to read; the `elif` gets lost.

Reply:
will do.